src/fec_rs_fnt.h

```diff
@@ -90,12 +90,19 @@ class RsFnt : public FecCode<T> {
 
         int m = arith::get_smallest_power_of_2<int>(n_data);
         this->fft = new fft::Radix2<T>(this->gf, n, m, pkt_size);
+
+        // vector stores r^{-i} for i = 0, ... , k
+        T inv_r = this->gf->inv(r);
+        this->inv_r_powers = new vec::Vector<T>(this->gf, this->n_data + 1);
```

Comment:
You could use a unique_ptr here, then you wouldn't need the delete in the destructor (it would be done automatically)

Reply:
Why using a raw pointer here?